deploy/operator/gather.sh

```diff
@@ -157,6 +157,11 @@ function gather_hypershift_data() {
   hypershift_dir="${LOGS_DEST}/hypershift"
   mkdir -p "${hypershift_dir}"
 
+  oc get po -n hypershift -o yaml > ${hypershift_dir}/hypershift_pods.yaml
+  oc get po -n hypershift -o custom-columns=:.metadata.name --no-headers | while read hypershift_pod_name ; do
+    oc logs -n hypershift $hypershift_pod_name > ${hypershift_dir}/hypershift_pod_log_$hypershift_pod_name.log
+  done
+
```

Comment:
I think we can delete line 165 to line 181

Reply:
> Also I don't get why it's trying to collect all [these](https://github.com/openshift/assisted-service/pull/5344/files#diff-9ff8483d183f18ac66f280b1002c37eba28d5d3be3078a02579738696e4ca3fcR169-R173) or [these](https://github.com/openshift/assisted-service/pull/5344/files#diff-9ff8483d183f18ac66f280b1002c37eba28d5d3be3078a02579738696e4ca3fcR175-R180) logs, assisted isn't running in this namespace, and all CRDs are already collected [elsewhere](https://gcsweb-ci.apps.ci.l2s4.p1.openshiftapps.com/gcs/origin-ci-test/pr-logs/pull/openshift_assisted-service/5341/pull-ci-openshift-assisted-service-release-ocm-2.7-e2e-ai-operator-ztp-capi/1677018019882078208/artifacts/e2e-ai-operator-ztp-capi/assisted-baremetal-operator-gather/artifacts/)

Oh so I think @danielerez added this for his [AI on zero worker hypershift enhancement](https://github.com/openshift/assisted-service/blob/master/docs/enhancements/ai-on-zero-worker-hypershift.md) ([PR link](https://github.com/openshift/assisted-service/pull/4577)). 

I think it could be relevant if we run tests that deploy AI on hypershift clusters, but maybe it's not as relevant for all hypershift tests.

Maybe these could be moved to a different function? What do you both think?